git-srep/src/main.rs

```diff
@@ -0,0 +1,184 @@
+use clap::Parser;
+use colored::Colorize;
+use git_utils_shared as git_utils;
+use rayon::prelude::*;
+use std::process::Command;
+
+/// Replace given phrase inside files of the provided paths.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// The word to substitute
+    search: String,
+    /// The word to substitute with
+    replace: String,
+
+    #[clap(
+        long,
+        short,
+        takes_value = false,
+        help = "Substituted word in git dirty files"
+    )]
+    force: bool,
+
+    #[clap(required = true)]
+    paths: Vec<String>,
+}
+
+fn is_file_dirty(file_path: &str) -> bool {
+    let modified = git_utils::file::file_is_modified(file_path);
+    let untracked = git_utils::file::file_is_untracked(file_path);
+    if modified {
+        println!("modified -> {}", file_path.red());
+    }
+    if untracked {
+        println!("Untracked -> {}", file_path.red())
+    }
+
+    modified || untracked
+}
+
+fn change_word_in_file(file_path: &str, search: &str, replace: &str) {
+    println!(
+        "Replacing {} -> {} in {}",
+        search,
+        replace,
+        file_path.green()
+    );
+    let change_success = Command::new("sed")
+        .args(["-i", &format!("s/{search}/{replace}/g"), file_path])
+        .status()
+        .expect("Failed ")
+        .success();
+
```

Comment:
When injecting content into `s/.../.../g`, you must escape as per rules of `sed` command.  For example, attempting to replace `foo/bar` would result in premature end of `sed` regex and cause an error or undesired behavior.

Reply:
Done in 2d21df4fb0695c3d7e101c27cc12cb7b5a86c45d